### DISCUSSION TITLE
Shelly plus 1 PM Support ? Anleitung

### DISCUSSION THREAD

**Discussion by Reborn2** [camueller/SmartApplianceEnabler#302] [Allgemein]
Hallo, hab eine Frage zu Shelly plus 1 PM

Unter der Seite 

http://192.168.0.229/rpc/Shelly.GetStatus

Gibt der 1 PM folgendes aus

{"ble":{},"cloud":{"connected":true},"input:0":{"id":0,"state":false},"mqtt":{"connected":false},"switch:0":{"id":0, "source":"init", "output":true, "apower":1160.1, "voltage":217.7, "current":5.007, "aenergy":{"total":6618.829,"by_minute":[15496.922,24463.525,24636.686],"minute_ts":1659619956},"temperature":{"tC":61.2, "tF":142.2}},"sys":{"mac":"441793CF4C2C","restart_required":false,"time":"15:32","unixtime":1659619959,"uptime":1286448,"ram_size":239472,"ram_free":166036,"fs_size":458752,"fs_free":217088,"cfg_rev":21,"available_updates":{"beta":{"version":"0.11.0-beta2"}}},"wifi":{"sta_ip":"192.168.0.229","status":"got ip","ssid":"IoT-2G","rssi":-37}}

mit $.status:0.apower lässt sich der Shelly somit auslesen. 

Könnte also in die Shelly doc übernommen werden.

**Comment by camueller** [comment]
Habe Deinen PR übernommen - vielen Dank!

**Comment by rob12345-new** [comment]
Hallo, 

erst eimal danke für die tolle Software!
Ich habe auch einen Shelly Plus 1PM für meine Pool Pumpe und teste damit gerade Parametersätze, daher jetzt auch der Post hier:
für das auslesen der Leistung muss der Pfad für Extraktion wie folgt lauten: 

**$.switch:0.apower** 

überprüfbar über: https://jsonpath.com/ 

wie man JSON extrahiert, wird hier ganz gut beschrieben:
https://goessner.net/articles/JsonPath/index.html#e2

Shelly gen2 API siehe hier: https://shelly-api-docs.shelly.cloud/gen2/ComponentsAndServices/Shelly#shellygetstatus-example

Eine Frage habe ich zu dem JSON Methode GET, POST, PUT, PATCH, DELETE -> wo finde ich eine Anleitung was ich damit machen kann?

Gruß

**Reply by camueller** [reply]
`GET`, `POST`, `PUT`, `PATCH`, `DELETE` sind Bestandteil des HTTP-Protokolls und haben nichts mit dem Inhalt der  übertragenen Daten zu tun. JSON wiederum ist ein Datenformat.

**Comment by alexb2506** [comment]
Ich habe hier wie folgt die Daten ausgelesen

URL: **http://ip-adresse/rpc/Switch.GetStatus?id=0**
Parameter: **LEISTUNG**
Pfad f. Extr.: **$.apower**

**Comment by alexb2506** [comment]
Hallo zusammen,
ich würde gerne Shelly Plug S sowie Shelly Plus 1PM für gleichzeitiges Messen und steuern einsetzen. Leider bekomme ich es nicht hin. Das Messen klappt sauber. SHM zeigt das Gerät als direkte Kommunikation an. Unter der Geräteübersicht tauchen die Shelly‘s nicht auf.
Was mache ich falsch?

**Reply by camueller** [reply]
Wenn die Shelly's in den SAE eingebunden sind, existieren sie aus Sicht des SHM bzw. Sunny Portals nicht. Der SAE nutzt sie lediglich zum Messen der Leistungsaufnahme und zum Schalten.

**Reply by alexb2506** [reply]
Folgendes habe ich konfiguriert:

Typ HTTP

URL * http://192.168.10.111/
Parameter / Aktion * Einschalten
Methode GET
Wert http://192.168.10.111/relay/0?turn=on
Umrechnungsfaktor 1

Das gleiche zum Ausschalten, lediglich mit Parameter OFF

Status über HTTP abfragen "aktiviert"

URL http://192.168.10.111/Status
Parameter Eingeschaltet
Methode GET
Daten http://192.168.10.111/relay/0?turn=on
Regex für Extraktion

Was habe ich hier falsch gemacht?

**Reply by camueller** [reply]
Du hast Wert und URL "verwurstet". Du brauchst keinen Wert und die URL muss so http://192.168.10.111/relay/0?turn=on sein.

**Reply by alexb2506** [reply]
Habe das jetzt so angepasst. Bekomme aber keine Veränderung im SHM. Wie kann ich die Zuschaltzeiten bei Überschuss definieren? Habe ich einen denkfehler? Muss ich Shelly neu Einlesen als Gerät?

![image](https://user-images.githubusercontent.com/114364041/194033520-fbb0d9dc-1323-47f1-a692-df4a09b98644.png)

![image](https://user-images.githubusercontent.com/114364041/194033382-8be0cc89-c124-44d9-8963-9ba9d54d481a.png)

**Reply by camueller** [reply]
> Zuschaltzeiten bei Überschuss definieren

Siehe Doku

> Muss ich Shelly neu Einlesen als Gerät?

nein

**Reply by alexb2506** [reply]
steht bei Shelly dazu nichts drin. im SHM kann ich es nicht nirgendswo einstellen.
Bitte um kurzen links zur richtigen Doku.
VG

**Comment by alexb2506** [comment]
Ich habe es nun doch Hinbekommen. Habe dazu als Kategorie die Wärmepumpe bzw. Pumpe gewählt. Spielt es eigentlich eine Rolle was Mann auswählt?

Zeitplan anlegen. Dann taucht es in der Konfiguration von SHM auf.

**Comment by runn8** [comment]
Hallo,
ich bräuchte bitte eure Hile.
Ich habe bereits 3 Shelly 1S im erfolgreich im Einsatz. Die URL für diese Geräte lautet http://IP-Adresse/meter/0
Beim Shelly 1PM funktioniert diese URL nicht. Die Status-URL http://IP-Adresse/status funktioniert einwandfrei, ebenso wie die beiden http://IP-Adresse/relay/0?turn=on und "off". Aber SAE benötigt die URL zum Stromverbrauch. Diese weiß ich und finde ich einfach nicht.

http://IP-Adresse/rpc/Shelly.GetStatus funktioniert ebenfalls nicht, bzw. kommt zurück: "Not found"

Hat hier vielleicht jemand eine Idee?

**Reply by camueller** [reply]
> http:///rpc/Shelly.GetStatus

> http:///meter/0

Das sind keine gültig URLs - die können nicht funktionieren! Außerdem kannst Du jede URL "testen", indem Du sie einfach in den Web-Browser eingibst. Wenn da nichts angezeigt wird, ist sie nicht gültig oder nicht richtig.

Halt Dich doch an die Doku - dafür gibt es die: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Shelly_DE.md

**Comment by runn8** [comment]
Hallo,
danke für deine Antwort! Github hat in meinem Beitrag meine spitzen Klammern + Inhalt entfernt, daher die falsche Darstellung. Habs ausgebessert. Es sieht folgendermaßen aus:

Shelly Plug - funktioniert bereits im SAE:
http://IP-Adresse/meter/0 - funktioniert
http://IP-Adresse/status - funktioniert
http://IP-Adresse/rpc/Shelly.GetStatus - funktioniert NICHT: "not found"

Shelly 1 PM ohne Plus - funktioniert bislang **nicht** im SAE:
http://IP-Adresse/meter/0 - funktioniert NICHT: "not found"
http://IP-Adresse/status - funktioniert
http://IP-Adresse/rpc/Shelly.GetStatus - funktioniert NICHT: "not found"

Der Shelly 1 PM benötigt demnach eine andere http-Adresse zur Leistungsabfrage. Diese habe ich bislang nicht gefunden. :-(
Da auch der rpc/Shelly.GetStatus Abfrage auf bereits funktionierenden Shellys nicht klappt, muss ich hier etwas ändern oder rpc aktivieren? Habe wirklich gesucht aber noch keine Antwort gefunden. 

Danke für jeden Tipp!

**Reply by camueller** [reply]
Die URL sind ja nicht SAE-spezifisch. Da sollte sich irgendwo (auf der Shelly-Homepage) die entsprechende Information finden lassen.

**Comment by chubakka0815** [comment]
Hallo,
ich habe meine Shelly 1 PM ( ohne Plus ) mit folgenden Einstellungen am laufen: 
![image](https://github.com/camueller/SmartApplianceEnabler/assets/121820466/9b794ee8-629a-4852-942f-0776c95b0482)

![image](https://github.com/camueller/SmartApplianceEnabler/assets/121820466/eb3bbc34-a012-42d5-9cb5-13be2cd05ea7)

Vielleicht kann es ja jemand gebrauchen

**Reply by runn8** [reply]
super Sache, so funktionierts! Vielen Dank!
wie bist du dahinter gekommen?

**Reply by chubakka0815** [reply]
Hab manchmal helle Momente 😄
Nee ganz im Ernst, hab probiert und getestet. Alle Varianten die mir in den Kopf gekommen sind, bzw hab mir die anderen Shelly s angesehen und dort das genommen was mir Ergebnisse geliefert hat
